### PR TITLE
Upgraded ResolveUnannotated

### PR DESCRIPTION
**Pull request by marc-jones**
Added @kkurets 's updates to ResolveUnannotated

### CODE REVIEW

File: workflows/preparegenome.nf [hunk 1]
```diff
@@ -52,18 +56,39 @@ workflow {
...
-    RESOLVE_UNANNOTATED(
-        ICOUNT_SEGMENT.out.regions, // segment
-        FILTER_GTF.out.post_filtering_gtf, // gtf
-        ch_fai, // fai
+    // Resolve unannotated for PEKA
+    RESOLVE_UNANNOTATED_GENIC_OTHER (
+        RAW_ICOUNT_SEGMENT.out.regions,      // filtered_segmentation
+        RENAME.out.renamed, // unfiltered_segmentation
+        FILTER_GTF.out.post_filtering_gtf,   // gtf
+        ch_fai,                              // fai
```
Comment (kkuret) [inline workflows/preparegenome.nf:85]: Hi @marc-jones! The --genic_other flag should be passed here, is this is done with the extended argument?

File: workflows/preparegenome.nf [hunk 2]
```diff
@@ -52,18 +56,39 @@ workflow {
...
 
+    RENAME (
+        FILTERED_ICOUNT_SEGMENT.out.regions,
+        "filtered.regions.gtf.gz"
+    )
+
+    // Resolve unannotated for iCount summary and other processes
+    RESOLVE_UNANNOTATED (
+        RAW_ICOUNT_SEGMENT.out.regions,      // filtered_segmentation
+        RENAME.out.renamed, // unfiltered_segmentation
```
Comment (kkuret) [inline workflows/preparegenome.nf]: are the raw segment and the filtered in the correct order here - filtered should be first I think? And why not RAW_ICOUNT_SEGMENT.out.regions and FILTERED_ICOUNT_SEGMENT.out.regions?

**Review by kkuret — COMMENTED** [review]
Had some questions when calling the resolve_unannotated process and added comments.

File: workflows/preparegenome.nf (hunk 1 above)
Comment (marc-jones) [reply workflows/preparegenome.nf:85]: That's passed in [here](https://github.com/goodwright/imaps-nf/pull/7/files#diff-e8c87a50ddb1e50082721ad74cf232c842aa8ca6a01e62c7c9a60880fdf0a4adR38). It's a bit messy but I think (@samirelanduk ?) that's the best way to do it.

File: workflows/preparegenome.nf (hunk 2 above)
Comment (marc-jones) [reply workflows/preparegenome.nf]: I think you're commenting on an old commit, as I replaced the `RENAME` module with a better solution. This was needed as both the `iCount segment` processes outputted `regions.gtf.gz` and there was an input name conflict.

Regarding the ordering, you are correct, I've got this the wrong way around, I'll push a fix.

Comment (kkuret) [reply workflows/preparegenome.nf]: cool!

File: workflows/preparegenome.nf (hunk 1 above)
Comment (samirelanduk) [reply workflows/preparegenome.nf:85]: Yeah haven't settled on a definitive way to pass params yet - this is absolutely fine for now.

**Review by samirelanduk — APPROVED** [review]
All looks great, and runs fine when run locally on iMaps. I'll merge this and deploy it to the live iMaps tonight.

**Pull request merged**